### PR TITLE
[FIX] Documentation links

### PR DESCRIPTION
**Pull request by rokgomiscek**
##### Issue
Some widgets reported missing documentation even if this wasn't the case.

##### Description of changes
Fixed names so documentation links point to the right file.

##### Includes
- [x] Code changes
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Orange/widgets/evaluate/owtestlearners.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ class State(enum.Enum):
 
 
 class OWTestLearners(OWWidget):
-    name = "Test & Score"
+    name = "Test and Score"
```
Comment (ajdapretnar) [inline Orange/widgets/evaluate/owtestlearners.py]: This changes the name of the widget in Orange. Which means all schemas that had "Test & Score" before could fail. Please check.

File: Orange/widgets/unsupervised/owkmeans.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ class NotEnoughData(ValueError):
 
 
 class OWKMeans(widget.OWWidget):
-    name = "k-Means"
+    name = "k-Means clustering"
```
Comment (ajdapretnar) [inline Orange/widgets/unsupervised/owkmeans.py]: This changes the name of the widget and makes it longer. I don't think this is ok. If docs don't match, we can change the docs.

File: doc/visual-programming/source/widgets/data/applydomain.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Transform
+Apply Domain
 =========
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/data/applydomain.md]: Underscores have to fit the length of the title.

File: doc/visual-programming/source/widgets/evaluation/confusionmatrix.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Predictions
+Confusion Matrix
 ===========
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/evaluation/confusionmatrix.md]: Same as above.

File: doc/visual-programming/source/widgets/data/applydomain.md (hunk 3 above)
Comment (ajdapretnar) [reply doc/visual-programming/source/widgets/data/applydomain.md]: Also, the change in not reflected in the index.rst. Could you please fix this too?

Comment (ajdapretnar) [reply doc/visual-programming/source/widgets/data/applydomain.md]: Sorry for these continuous comments. The documentation for Apply Domain should be changed to reflect the changes in the widget name. Namely the references to the widget, image names (so we know what they refer to) and likely the example workflows.

File: Orange/widgets/evaluate/owtestlearners.py (hunk 1 above)
Comment (rokgomiscek) [reply Orange/widgets/evaluate/owtestlearners.py]: It works (I checked on an example workflow).

File: Orange/widgets/unsupervised/owkmeans.py (hunk 2 above)
Comment (rokgomiscek) [reply Orange/widgets/unsupervised/owkmeans.py]: Changing the name would make this similar to Hierachical and Louvain Clustering (though "clustering" should be capitalized).

File: doc/visual-programming/source/widgets/data/applydomain.md (hunk 3 above)
Comment (rokgomiscek) [reply doc/visual-programming/source/widgets/data/applydomain.md]: Done.
Apply Domain (ex Transform) not present in example workflows.

File: Orange/widgets/unsupervised/owkmeans.py (hunk 2 above)
Comment (lanzagar) [reply Orange/widgets/unsupervised/owkmeans.py]: Hierarchical and Louvain can not be used by themselves - "clustering" is part of the name there.
k-Means is clear enough and can be used as a name by itself.
I think we should strive for short widget names, so I would not expand it in this case.
BTW, clustering is already a keyword for k-means so using it in a search lists it alongside other methods.

Comment (rokgomiscek) [reply Orange/widgets/unsupervised/owkmeans.py]: Ok. Changed back widget name and changed documentation file name.

File: doc/visual-programming/source/widgets/data/applydomain.md (hunk 3 above)
Comment (ajdapretnar) [reply doc/visual-programming/source/widgets/data/applydomain.md]: Sorry, I didn't mean example workflows in Orange tutorial section, but workflows that are in Example section of the documentation.

**Pull request merged**